File: Leetcode/Minimum Number of Arrows to Burst Balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, Comparator.comparingInt(point -> point[1]));
        int minArrowShots = 1;
        int minRightX = points[0][1];
        for (var currPoint : points) {
            if (minRightX < currPoint[0]) {
                ++minArrowShots;
                minRightX = currPoint[1];
            }
        }
        return minArrowShots;
    }
}

File: Leetcode/Minimum Number of Arrows to Burst Balloons.js
function findMinArrowShots(points) {
    return points.sort((a, b) => a[1] - b[1])
                 .reduce(getArrowsShotMaxEnd, {arrowsShot: 1, maxEnd: points[0][1]})
                 .arrowsShot;
}

function getArrowsShotMaxEnd({arrowsShot, maxEnd}, point) {
    if(maxEnd < point[0]) {
        ++arrowsShot;
        maxEnd = point[1];
    }
    return {arrowsShot, maxEnd};
}
